overlord/servicestate/service_control_test.go

```diff
@@ -127,6 +127,22 @@ func (s *serviceControlSuite) mockTestSnap(c *C) *snap.Info {
 	return info
 }
 
+func verifyUnsortedInvocations(c *C, sysctlArgs [][]string, startIndex int,
+	action string, expectedArguments []string) {
+	/* We don't care about the order of the is-enabled invocations, as long as
+	 * they all happen before the first invocation of "start" */
+	arguments := []string{}
+	for _, params := range sysctlArgs[startIndex:] {
+		if params[0] == action {
+			arguments = append(arguments, params[1])
+		} else {
+			break
+		}
```

Comment:
I think this could be simplified if the function was passed a slice of first :N elements per my comment below to line 633.

Alternatively:
```
if params[0] != action {
  break
}
arguments = append(arguments, params[1])
```

Reply:
Reworked the code a bit.